diff --git a/agent/finetune/train_ppo_diffusion_img_agent.py b/agent/finetune/train_ppo_diffusion_img_agent.py
index 9eb6696..a57d035 100644
--- a/agent/finetune/train_ppo_diffusion_img_agent.py
+++ b/agent/finetune/train_ppo_diffusion_img_agent.py
@@ -171,12 +171,10 @@ class TrainPPOImgDiffusionAgent(TrainPPODiffusionAgent):
             if not eval_mode:
                 with torch.no_grad():
                     # apply image randomization
-                    obs_trajs["rgb"] = (
-                        torch.from_numpy(obs_trajs["rgb"]).float().to(self.device)
-                    )
-                    obs_trajs["state"] = (
-                        torch.from_numpy(obs_trajs["state"]).float().to(self.device)
-                    )
+                    for obs_key in obs_trajs:
+                        obs_trajs[obs_key] = (
+                            torch.from_numpy(obs_trajs[obs_key]).float().to(self.device)
+                        )
                     if self.augment:
                         rgb = einops.rearrange(
                             obs_trajs["rgb"],
